HDSAviz/interactivevislib.py

```diff
@@ -0,0 +1,186 @@
+"""
+The current program is meant to draw interactive visualizations using BOKEH
+and ipywidgets libraries for sensitivity analysis
+
+Dependencies :
+Currently this code needs to be run Inside Ipython notebook.
+Python version :2.7
+
+Note:
+Please ensure Basic_Plot.py exists in same directory and necessary data files are
+located in c:/HDSAviz_data folder
+
+Basically files should be organized as :
+
+C: HDSAviz_data/
+C: HDSAviz_data/analysis_CO.txt
+C: HDSAviz_data/analysis_CO2.txt
+etc.
+
+C: HDSAviz_data/interactivevislib.py
+C: HDSAviz_data/Interactive_vis_toolkit.ipynb
+
+"""
+
+#import Basic_Plot  once Basic_Plot.py file is ammended.
+
+import os
+from bokeh.models.widgets import Panel, Tabs
+from bokeh.io import output_file, output_notebook, show, vform
+from ipywidgets import FloatSlider
+from IPython.html.widgets import interact
+from collections import OrderedDict
+from bokeh.plotting import figure, show, output_notebook, output_file
+
+# initializing the range of values for total order (ST) and first order
+# (S1) sensitivity tests
+ST_Range = 1
+S1_Range = 1
+
+def change_directory():
+    """
+    This function changes the working directory from C:Users/UserName to
+    C:/HDSAviz_data/ which is chosen as convention for current project.
+
+    Output:
+    Changes working directory and prints current directory
+    """
+    path = "c:/HDSAviz_data"
+    os.chdir(path)
+    print 'current working directory: ' + os.getcwd()
+
+
+def gather_all_analysis_files():
+    """
+    This function gathers all the analysis files from C:/HDSAvis_data/
+
+    Output:
+    Returns a list of analysis files that can be fed into plotting function
+
+    """
+    new = !ls
+    analysis_files = []
+    for files in new:
+        if files.startswith("analysis"):
+            analysis_files.append(files)
+    print 'all analysis files gathered from current directory!'
+    return analysis_files
+
+def plot_all_outcomes_burtin(S1_Range,ST_Range):
```

Comment:
I think it can be done using save button in bokeh plots... i sent out some hints regarding hat via import previewsave tool and i also incorporated some hard coded saving options for files. see demo.html and multitab_sensitivity_plots.html